src/psyclone/tests/psyGen_test.py

```diff
@@ -3756,6 +3819,47 @@ def test_fparser2astprocessor_handling_binaryopbase(f2008_parser):
     assert new_node._operator == '+'
 
 
+def test_fparser2astprocessor_handling_unaryopbase(f2008_parser):
+    ''' Test that fparser2 UnaryOpBase is converted to the expected PSyIR
+    tree structure.
+    '''
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Execution_Part, UnaryOpBase
+    reader = FortranStringReader("x=-4")
+    fparser2unary_operation = Execution_Part.match(reader)[0][0].items[2]
+    assert isinstance(fparser2unary_operation, UnaryOpBase)
+
+    fake_parent = Node()
+    processor = Fparser2ASTProcessor()
+    processor.process_nodes(fake_parent, [fparser2unary_operation], None)
+    # Check a new node was generated and connected to parent
+    assert len(fake_parent.children) == 1
+    new_node = fake_parent.children[0]
+    assert isinstance(new_node, UnaryOperation)
+    assert len(new_node.children) == 1
+    assert new_node._operator == '-'
+
+
+def test_fparser2astprocessor_handling_return_stmt(f2008_parser):
+    ''' Test that fparser2 Return_Stmt is converted to the expected PSyIR
+    tree structure.
+    '''
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Execution_Part, Return_Stmt
+    reader = FortranStringReader("return")
+    return_stmt = Execution_Part.match(reader)[0][0]
+    assert isinstance(return_stmt, Return_Stmt)
+
+    fake_parent = Node()
+    processor = Fparser2ASTProcessor()
+    processor.process_nodes(fake_parent, [return_stmt], None)
+    # Check a new node was generated and connected to parent
+    assert len(fake_parent.children) == 1
+    new_node = fake_parent.children[0]
+    assert isinstance(new_node, Return)
+    assert len(new_node.children) == 0
```

Comment:
pylint prefers `assert not new_node.children`

Reply:
fixed